src/client/pages/OfferNew/index.tsx

```diff
@@ -59,18 +60,48 @@ export const OfferNew: React.FC = () => {
     skip: quoteIdsIsLoading,
   })
 
+  const bundleVariants = React.useMemo(
+    () =>
+      (data?.quoteBundle.possibleVariations as Array<QuoteBundleVariant>) ?? [],
+    [data?.quoteBundle.possibleVariations],
+  )
+  const canShowInsuranceSelector =
+    isInsuranceToggleEnabled && bundleVariants.length > 1
+  const [selectedBundleVariant, setSelectedBundleVariant] = React.useState<
+    QuoteBundleVariant
+  >()
+
+  const updateBundleVariant = (
+    bundleVariations: QuoteBundleVariant[],
+    selectedBundleVariant?: QuoteBundleVariant,
+  ) => {
+    if (!selectedBundleVariant) {
+      return setSelectedBundleVariant(bundleVariations[0])
+    }
+
+    const matchingVariant = bundleVariations.find(
+      ({ id }) => id === selectedBundleVariant.id,
+    )
+    if (!matchingVariant) return setSelectedBundleVariant(bundleVariations[0])
+    return setSelectedBundleVariant(matchingVariant)
+  }
+
+  useEffect(() => {
+    updateBundleVariant(bundleVariants, selectedBundleVariant)
+  }, [bundleVariants])
```

Comment:
This will fail "exhaustive deps" linting rule, no? I prefer writing `useEffect` with all dependencies listed.